### PR TITLE
Merge flesh monsters

### PR DESCRIPTION
**Pull request by JosephCarrington**
#### Summary
SUMMARY: Content "Add Flesh Monsters to Jabberwock monster track."

#### Purpose of change
This PR creates two new monsters, who will hopefully eventually populate the Industrial Sectors.

#### Describe the solution
In addition to the two new monsters on the Jabberwock track, this PR adds a absorb_meat special attack, which heals the monster that absorbs the meat, and can even `overheal` it. The maximum meat absorbed by the monster is 1/10th the monster's volume, to avoid monsters absorbing full corpses (unless the monster happens to be very large).

#### Additional context
In a (hopefully) upcoming PR I intend to add some Industrial Buildings where these monsters might spawn, including a meatpacking plant and a slaughterhouse. These locations should give the flesh monsters plenty of fodder to grow.

### CODE REVIEW

**Comment by JosephCarrington** [comment]
I should note that many of the numbers of the monster attributes are just guesses on my part as to what they should be. I do not yet really understand what they mean

**Comment by Ilysen** [comment]
In regards to attributes: both monsters are slow. Consider that a default human's turns is 100 moves; a player will be able to outwalk these monsters. If you want them to be slow, horrifying, lumbering threats, then the statline you have is just right, but if you want them to be very fast like the jabberwock, you should bump up the speed.

They're also pretty fragile. 100 health is just above that of a regular zombie, and since they have low armor and no dodging skill, they'd die pretty quickly. Zombie hulks have 480 health, if you're looking for an example; jabberwocks have 400.

Both of their melee damage is *nasty.* 6d4 is no laughing matter, especially considering that a character with 8 strength and no traits only has 84 HP on their limbs.

I'd be fine with what's there right now, since it gives an interesting niche; a monster that's slow and weak, but devastating in close combat, like the incandescent husk. But, if you'd like to make them fill a certain role, I hope that my advice can help with that.

**Comment by Night-Pryanik** [comment]
Jenkins rebuild

File: src/monattack.cpp [hunk 1]
```diff
@@ -3797,6 +3797,59 @@ bool mattack::flesh_golem( monster *z )
     return true;
 }
 
+bool mattack::absorb_meat( monster *z )
+{
+    //Absorb no more than 1/10th monster's volume, times the volume of a meat chunk
+    const int monster_volume = units::to_liter( z->get_volume() );
+    const float average_meat_chunk_volume = 0.5;
+    //TODO: dynamically get volume of meat
+    const int max_meat_absorbed = ( monster_volume / 10 ) * average_meat_chunk_volume;
```
Comment (Night-Pryanik) [inline src/monattack.cpp]:
```suggestion
    const int max_meat_absorbed = monster_volume / 10 * average_meat_chunk_volume;
```

File: src/monattack.cpp [hunk 2]
```diff
@@ -3797,6 +3797,59 @@ bool mattack::flesh_golem( monster *z )
...
+                             current_item.tname() );
+                    return false;
+                }
+                if( current_item.count_by_charges() ) {
+                    //Choose a random amount of meat charges to absorb
+                    long meat_absorbed = std::min( max_meat_absorbed, rng( 1, total_charges ) );
+                    const int hp_to_heal = meat_absorbed * ml_per_charge * meat_absorption_factor;
+                    z->heal( hp_to_heal, true );
+                    g->m.use_charges( p, 0, current_item.type->get_id(), meat_absorbed );
+
```
Comment (Night-Pryanik) [inline src/monattack.cpp]:
```suggestion
```

File: src/monattack.cpp [hunk 3]
```diff
@@ -3797,6 +3797,59 @@ bool mattack::flesh_golem( monster *z )
...
+                } else {
+                    //Only absorb one meaty item
+                    long meat_absorbed = 1;
```
Comment (Night-Pryanik) [inline src/monattack.cpp]: I think we could avoid initializing this variable and just use `1` where needed below.

File: src/monattack.cpp [hunk 4]
```diff
@@ -3797,6 +3797,59 @@ bool mattack::flesh_golem( monster *z )
...
+                    const int hp_to_heal = meat_absorbed * ml_per_charge * meat_absorption_factor;
+                    z->heal( hp_to_heal, true );
+                    g->m.use_amount( p, 0, current_item.type->get_id(), meat_absorbed );
+                }
+                if( g->u.sees( *z ) ) {
+                    add_msg( m_warning, _( "The %1$s absorbs the %2$s, growing larger." ), z->name(),
+                             current_item.tname() );
+                    add_msg( m_debug, _( "The %1$s now has %2$s out of %3$s hp" ), z->name(), z->get_hp(),
```
Comment (Night-Pryanik) [inline src/monattack.cpp]: No need to translate debug messages.
```suggestion
                    add_msg( m_debug, "The %1$s now has %2$s out of %3$s hp", z->name(), z->get_hp(),
```

**Comment by JosephCarrington** [comment]
The 100 and 200 base HP seem to go higher, I've seen a flesh golem get up to about 300hp after absorbing a bunch of butchery refuse. Theres also the fact that if you happen to be fighting one next to a pile of meaty stuff, it's basically constantly healing away the  damage you inflict on it. The jabberwock is pretty darn fast, and since these monsters are on that track, I will make the flesh golem slightly faster than it currently is.

File: src/monattack.cpp (hunk 3 above)
Comment (JosephCarrington) [reply src/monattack.cpp]: It turns out this has to be a long, in order that the heal function can decrement it... fun!

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

**Pull request merged**